common/network-common/src/main/java/org/apache/spark/network/util/TransportFrameDecoder.java

```diff
@@ -139,14 +139,18 @@ private ByteBuf decodeNext() throws Exception {
       return nextBufferForFrame(remaining);
     }
 
-    // Otherwise, create a composite buffer.
-    CompositeByteBuf frame = buffers.getFirst().alloc().compositeBuffer();
```

Comment:
actually we can set the maxNumComponents for `compositeBuffer` to avoid consolidate underlying, such as `CompositeByteBuf frame = buffers.getFirst().alloc().compositeBuffer(Integer.MAX_VALUE);`, but this might be not a good choice.